src/mqueryfront/src/QueryField.js

```diff
@@ -31,18 +32,34 @@ class QueryField extends Component {
         return this.state.selectedTaint;
     }
 
-    componentWillReceiveProps(newProps) {
-        this.setState({
-            rawYara: newProps.rawYara,
-            isLocked: newProps.isLocked,
-        });
+    static getDerivedStateFromProps(nextProps, prevState) {
+        if (
+            nextProps.rawYara !== prevState.rawYara ||
+            nextProps.readOnly !== prevState.readOnly
+        ) {
+            return { rawYara: nextProps.rawYara, readOnly: nextProps.readOnly };
+        }
+    }
```

Comment:
Nice - I knew we're doing it in an obsolete way, but never was motivated enough to refactor it.